src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs

```diff
@@ -471,6 +471,31 @@ public static IEnumerable<object[]> GetAsync_IPBasedUri_Success_MemberData()
             }
         }
 
+        [Fact]
+        public async Task GetAsync_IPv6AddressInHostHeader_CorrectlyFormatted()
+        {
+            string host = "[::1234]:8080";
+            string ipv6Address = "http://" + host;
+            bool connectionAccepted = false;
+
+            await LoopbackServer.CreateClientAndServerAsync(async proxyUri =>
+            {
+                using (HttpClientHandler handler = CreateHttpClientHandler())
+                using (var client = new HttpClient(handler))
+                {
+                    handler.Proxy = new WebProxy(proxyUri);
```

Comment:
Clever

Reply:
Actually you [taught me this](https://github.com/dotnet/corefx/pull/28150#issuecomment-373914546). : )

Reply:
Heh, ok, at least I'm consistent :)